code/datums/smells/smell.dm

```diff
@@ -0,0 +1,88 @@
+#define SMELL_HORRIBLE 0
+#define SMELL_FOUL 1
+#define SMELL_NEUTRAL 2
+#define SMELL_PLEASANT 3
+#define SMELL_NICE 4
+
+#define SMELL_TRACE 2
+#define SMELL_FAINT 4
+#define SMELL_NORMAL 8
+#define SMELL_INTENSE 16
+
+/datum/smell
+	var/qty
+	var/name
+	var/intensity
+	var/description
+
+/datum/smell/New(var/q)
```

Comment:
What does `q` mean? I can't tell what `qty` or `q` means because they are tiny names that mean nothing to anyone but the guy who programmed them. Add proper argument names.